samples/client/petstore/python-asyncio/tests/test_api_client.py

```diff
@@ -0,0 +1,27 @@
+# coding: utf-8
+
+# flake8: noqa
+
+import unittest
+import weakref
+
+from tests.util import async_test
+import petstore_api
+
+
+class TestApiClient(unittest.TestCase):
+
+    @async_test
+    def test_context_manager_closes_client(self):
+
+        with petstore_api.ApiClient() as client:
+            # thread pool
+            self.assertIsNotNone(client.pool)
+            pool_ref = weakref.ref(client._pool)
+            self.assertIsNotNone(pool_ref())
+            # pool_manager
+            self.assertFalse(client.rest_client.pool_manager.closed)
+            rest_pool_ref = client.rest_client.pool_manager
+
+        self.assertIsNone(pool_ref())
+        self.assertFalse(rest_pool_ref.closed)
```

Comment:
When do we check that the pool_manager is closed?

Reply:
Thanks for spotting this. I had to switch to the asynchronous context generator to make it works. Please take a look again.